### PR TITLE
use vendor libraries from npm and various fixes

### PR DESCRIPTION
**Pull request by haarg**
This removes the vendored libraries we had, and instead uses them from npm.

It also fixes various problems that were introduced with the conversion to esbuild. With that conversion, the jQuery object and other functions were no longer globally available. This broke various bits of inline code.

This removes all of the inline code, like `onclick` handlers.

### CODE REVIEW

**Comment by haarg** [comment]
This should fix #3062 and #3065

File: root/static/js/cpan.js [hunk 1]
```diff
@@ -325,126 +227,97 @@
...
+            alert('Please sign in to add favorites');
+        });
+    }
 
-    $(document).on("dragenter", function(event) {
-        renderer.addClass("dragging");
-    });
+    for (const sel of document.querySelectorAll('.select-navigator')) {
+        sel.addEventListener('change', () => {
+            document.location.href = sel.value;
```
[the remark this answers is not in the document]
Comment (haarg) [reply root/static/js/cpan.js:288]: In the way this is implemented, it's no more vulnerable than an `<a href="...">`.

**Comment by oalders** [comment]
@dhogan8 if you have a moment, can you take a *quick* look at the front end changes? The diff is too large for a human to handle, but even if you spent 5 or 10 minutes on it, that would be helpful.

File: Dockerfile [hunk 2]
```diff
@@ -41,7 +41,7 @@ WORKDIR /app/
 COPY cpanfile cpanfile.snapshot ./
 RUN \
     --mount=type=cache,target=/root/.perl-cpm,sharing=private \
-<<EOT /bin/bash -euo pipefail
+<<EOT
```
Comment (oalders) [inline Dockerfile:44]: Wondering why the bash strict mode is getting removed?

**Review by oalders — APPROVED** [review]

File: Dockerfile (hunk 2 above)
Comment (haarg) [reply Dockerfile:44]: It's already set with a `SHELL` command. I've gone back and forth on how I wanted to handle this, but for now the `SHELL` command seems fine.

**Pull request merged**